### ISSUE TITLE
Ops Manager EKS Sharded deployment is failing 

### ISSUE THREAD

**Issue by dileep8014** [mongodb/mongodb-enterprise-kubernetes#47]
i am creating mongo deployment using Mongodb Kubernetes Enterprise Operator inside AWS EKS with TLS enabled, All the clusters are running but i see the below error inside the agent logs while deploying and tis failing after downloading the mongo db, 

any help would be appreciated

```
 Error determining if process=mongos-cluster-alpha-0-1.mongos-cluster-alpha-sh.mongodb.svc.cluster.local:27017 (local=false) is up : <mongos-cluster-alpha-0-1> [23:37:12.628] Error executing WithClientFor() for cp=mongos-cluster-alpha-0-1.mongos-cluster-alpha-sh.mongodb.svc.cluster.local:27017 (local=false) connectMode=SingleConnect : <mongos-cluster-alpha-0-1> [23:37:12.628] Error checking out client (0x0) for connParam=mongos-cluster-alpha-0-1.mongos-cluster-alpha-sh.mongodb.svc.cluster.local:27017 (local=false) connectMode=SingleConnect : [23:37:12.628] Error dialing to connParams=mongos-cluster-alpha-0-1.mongos-cluster-alpha-sh.mongodb.svc.cluster.local:27017 (local=false): tried 4 identities, but none of them worked. They were (CN=mongos-cluster-alpha-0-1,OU=MongoDB Kubernetes Operator,O=mongodb,L=NY,ST=NY,C=US@$external[[MONGODB-X509]][0], CN=mms-automation-agent,OU=MongoDB Kubernetes Operator,O=mms-automation-agent,L=NY,ST=NY,C=US@$external[[MONGODB-X509]][0], CN=mongos-cluster-alpha-0-1,OU=MongoDB Kubernetes Operator,O=mongodb,L=NY,ST=NY,C=US@$external[[MONGODB-X509]][0], )

```

**Comment by dileep8014** [comment]
below is the detailed error - 

github.com/10gen/ops-manager-kubernetes/pkg/controller/operator.(*errorStatus).updateStatus
	/data/mci/947534fb5d13d8b7f9314f005fd10f61/src/github.com/10gen/ops-manager-kubernetes/pkg/controller/operator/reconcileresult.go:103
github.com/10gen/ops-manager-kubernetes/pkg/controller/operator.(*ReconcileMongoDbShardedCluster).Reconcile
	/data/mci/947534fb5d13d8b7f9314f005fd10f61/src/github.com/10gen/ops-manager-kubernetes/pkg/controller/operator/mongodbshardedcluster_controller.go:48
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/data/mci/947534fb5d13d8b7f9314f005fd10f61/src/github.com/10gen/ops-manager-kubernetes/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:216
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/data/mci/947534fb5d13d8b7f9314f005fd10f61/src/github.com/10gen/ops-manager-kubernetes/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:192
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	/data/mci/947534fb5d13d8b7f9314f005fd10f61/src/github.com/10gen/ops-manager-kubernetes/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:171
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/data/mci/947534fb5d13d8b7f9314f005fd10f61/src/github.com/10gen/ops-manager-kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/data/mci/947534fb5d13d8b7f9314f005fd10f61/src/github.com/10gen/ops-manager-kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	/data/mci/947534fb5d13d8b7f9314f005fd10f61/src/github.com/10gen/ops-manager-kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88

**Comment by antonlisovenko** [comment]
hi @dileep8014 
sorry for late response, is this still an issue?

**Comment by dileep8014** [comment]
@antonlisovenko - i am still getting this error with TLS enabled

**Comment by rodrigovalin** [comment]
Hello @dileep8014 

Can you paste the output from `kubectl -n mongodb get mdb/mongos-cluster-alpha -o yaml` please?
Also, what version of the operator are you using?

Are the TLS certificates being created automatically by the operator? We know of an issue with the operator 1.4.1 and previous versions where the TLS certificates will not be created correctly in EKS. This has been fixed in 1.4.2. Can you try that version and tell us the results?

Additionally to this, if the certificates have been generated already, you will want to remove them, and the "Secret" object containing them.

**Comment by theburi** [comment]
@dileep8014 we haven't heard back on this issue. We have released many updates and I suspect this could not be a problem anymore. Could you let us know if this is still an issue?
